lib/Dialect/XTenNN/IR/XTenNNOps.cpp

```diff
@@ -1061,3 +1063,21 @@ LogicalResult ReduceMeanOp::inferReturnTypeComponents(
       ShapedTypeComponents(outputShape, inTy.getElementType()));
   return success();
 }
+
+//===----------------------------------------------------------------------===//
+// LoadExternalConstOp
+//===----------------------------------------------------------------------===//
+
+void LoadExternalConstOp::build(mlir::OpBuilder &odsBuilder,
+                                mlir::OperationState &odsState,
+                                mlir::TypeRange resultTypes,
+                                llvm::StringRef key, llvm::StringRef file) {
+  assert(resultTypes.size() == 1 &&
+         "LoadExternalConstOp must have exactly one result");
+  auto keyAttr = StringAttr::get(odsBuilder.getContext(), key);
+  auto fileAttr = StringAttr::get(odsBuilder.getContext(), file);
+  // Use the auto-generated builder that takes TypeRange, StringAttr,
+  // StringAttr, optional StringAttr
+  LoadExternalConstOp::build(odsBuilder, odsState, resultTypes, keyAttr,
+                             fileAttr, /*method=*/nullptr);
```

Comment:
Wouldn't it perhaps be better to use `std::nullopt` as `OptionalAttr` is a `std::optional`?
Although it also works the way you had it.
```suggestion
                             fileAttr, /*method=*/std::nullopt);
```

```
class OptionalAttr<Attr attr> : Attr<attr.predicate, attr.summary> {
  ...
  let returnType = "::std::optional<" # attr.returnType #">";
```